packages/core/src/Stores/ui-store.js

```diff
@@ -671,4 +671,21 @@ export default class UIStore extends BaseStore {
     showAccountTypesModalForEuropean() {
         this.toggleAccountTypesModal(this.root_store.client.is_uk);
     }
+
+    @action.bound
+    notifyAppInstall(prompt) {
+        this.deferred_prompt = prompt;
+        setTimeout(() => {
+            this.addNotificationMessageByKey('install_pwa');
+        }, 10000);
+    }
+
+    @action.bound
+    async installWithDeferredPrompt() {
+        this.deferred_prompt.prompt();
+        const choice = await this.deferred_prompt.userChoice;
```

Comment:
Are the properties (`userChoice`, `outcome`) always present here? No chance for an `undefined`?

Reply:
Yes. There should be no case of `undefined`. The promise just won't resolve.